packages/lesswrong/components/books/BookAnimation.tsx

```diff
@@ -0,0 +1,233 @@
+      background: 'transparent',
+      height: '300px',
+      width: 'calc(var(--half-spine-width, 14px)*2)',
+      position: 'absolute',
+      left: 'var(--negative-half-spine-width, -14px)',
+      transform: 'rotateY(-90deg) translateX(var(--negative-half-spine-width, -14px))'
+    },
+    '& .book > .spine-thickness': {
+      position: 'absolute',
+      height: '300px',
+      width: '1px',
+      background: 'white',
+      transform: 'translateZ(-1px) translateX(1px) rotateY(-45deg)'
+    },
+    '& .spine::after': {
+      content: '""',
+      width: '3px',
+      background: 'white'
+    },
+    '& .book::after': {
+      content: '""',
+      position: 'absolute',
+      left: '0',
+      width: '200px',
+      height: '300px',
+      'border-top-right-radius': '3px',
+      'border-bottom-right-radius': '3px',
+      'background': 'white',
+      'transform': 'translateZ(var(--negative-spine-width, -26px))',
+      'box-shadow': '5px 0 40px 9px #BBB',
+      'z-index': '-1',
+      'transition': 'opacity 1s ease'
+    },
+    '& .epistemology': {
+      '--starting-position': '0px',
+      '--collapsed-position': '8px',
+      '--negative-spine-width': '-19.166px',
+      '--half-spine-width': '8.5833px',
+      '--negative-half-spine-width': '-8.5833px',
+    },
+
+    '& .agency': {
+      '--starting-position': '230px',
+      '--collapsed-position': '27.166px',
+      '--negative-spine-width': '-13px',
+      '--half-spine-width': '6.5px',
+      '--negative-half-spine-width': '-6.5px'
+    },
+
+    '& .coordination': {
+      '--starting-position': '460px',
+      '--collapsed-position': '55.33px',
+      '--negative-spine-width': '-20.166px',
+      '--half-spine-width': '10.083px',
+      '--negative-half-spine-width': '-10.083px'
+    },
+
+    '& .curiosity': {
+      '--starting-position': '690px',
+      '--collapsed-position': '81.66px',
+      '--negative-spine-width': '-18.333px',
+      '--half-spine-width': '9.166px',
+      '--negative-half-spine-width': '-9.166px'
+    },
+
+    '& .alignment': {
+      '--starting-position': '920px',
+      '--collapsed-position': '110.82px',
+      '--negative-spine-width': '-21.166px',
+      '--half-spine-width': '10.583px',
+      '--negative-half-spine-width': '-10.583px',
+    },
+
+    '& .outer-wrapper': {
+
+    },
+
+    '& .hidden-quote': {
+      'opacity': '0%',
+      'transition': 'opacity 1s',
+      'position': 'absolute',
+      'top': '20%',
+      'left': '50%',
+      'z-index': '1',
+    },
+
+    '& .parent-container:hover .hidden-quote': {
+      'opacity': '100%'
+    }
+  },
+  revealedContent: {
+    opacity: 0,
+    transition: 'opacity 1s ease',
+    animationDelay: '1s'
+  }
+})
+
+const BookAnimation = ({ classes, children }: {
+  classes: ClassesType,
+  children: any
+}) => {
+
+  return (
+    <div className={classes.root}>
+      <div className="parent-container">
+        <div className="book-container epistemology">
```

Comment:
Convention-note: we don't usually use string classnames (as opposed to JSS generated classnames). So we would write this line as
```
<div className={classNames(classes.bookContainer" classes.epistemology)}>
```
and adjust the `styles` block in the same way. To use a JSS classname inside a complex JSS selector, you prepend a `$`, so for example the line
```
'& .parent-container:hover .book-container': {
```
would become
```
'& $parentContainer:hover $bookContainer': {
```

Reply:
Yep, the reason for this is just that I built this whole hacky thing in codepen, and then copied it over. Agree that we should move everything to proper JSS as soon as we have some Slack.